rero_ils/dojson/utils.py

```
@@ -212,9 +242,58 @@ def get_subfields(self, field, code=None):
             raise ValueError('data field expected (tag >= 01x)')
         return subfields
 
+    def build_value_with_alternate_graphic(
+            self, tag, code, label, index, link,
+            punct=None, spaced_punct=None):
+        """
+        Build the data structure for alternate graphical representation.
+
+        :param tag: the marc field tag
+        :param code: contains the subfield code. Used for debug only
+        :param label: the subfield data value
+        :param index: the subfield index position in the field
+        :param link: the link code to the alternate graphic field 880
+        :param punct: punctuation chars to remove i.e. ',.'
+        :param spaced_punct: punctuation chars preceded by a space to remove
+        :return: a list of 1 value, or 2 values if alternate graphical exists
+        :rtype: list
+
+        Example of return value:
+        [
+            {
+                "value": "B.I. Bursov"
+            },
+            {
+                "value": "Б.И. Бурсов",
+                "language": "rus-cyrl"
+            }
+        ]
+        """
+        def clean_punctuation(value, punct, spaced_punct):
+            return remove_trailing_punctuation(
+                value,
+                punctuation=punct,
+                spaced_punctuation=spaced_punct)
+
+        # build_value_with_alternate_graphic starts here
+
+        data = [{
+            'value': clean_punctuation(label, punct, spaced_punct).strip()
+        }]
+        try:
+            alt_gr = self.alternate_graphic[tag][link]
+            subfield = self.get_subfields(alt_gr['field'])[index]
+            data.append({
+                'value': clean_punctuation(subfield, punct, spaced_punct),
+                'language': self.get_language_script(alt_gr['script'])
+            })
+        except Exception as err:
```

Really? Is it not too dangerous to ignore the errors?